docs/RFCS/20222202_historical_hot_ranges.md

```diff
@@ -0,0 +1,224 @@
+
+HHR will be visualized by a heatmap in DB Console. The heatmap will be similar in spirit to what is offered by Google Cloud's (Big Table and Spanner) Key Visualizer, and TiDB's key visualizer. The heatmap will visualize range hotness across time and keyspace.
+
+Finally, the aggregated hot range data will be exported as a Prometheus metric, for self-hosted customers to consume with their own monitoring infrastructure. This has the benefit of providing redundancy, where the metrics remain available even if the cluster and DB console are unreachable.
+
+
+## Motivation
+Currently, CRDB does not have an out-of-the-box solution to provide users with access to historical values of their cluster's hottest ranges. Hot range related performance issues are often transient, so without access to historical data, a customer must observe their cluster's hot range data in real time. This requirement is not realistic, so the ultimate goal of this RFC is to provide a solution that remedies this.
+
+
+## Non-Goals of this RFC
+There are a few items that will be considered out of scope for this RFC.
+
+1. Changing the definition and/or computation of the hot range metric, despite QPS having known issues.
+    https://github.com/cockroachdb/cockroach/issues/50620<br/>
+    https://github.com/cockroachdb/cockroach/issues/73731
+
+2. It is not an explicit goal to showcase a metric-agnostic aggregation, storage, and visualization scheme. However, parts of this RFC may be applicable to other high cardinality metrics we surface in the future. Future observability work my find this RFC helpful.
+
+Importantly, none of the decisions being made in this proposal are irreversable, and can be changed in the future.
+
+
+## Detailed Design
+
+### Storage Requirements
+If we're not careful, HHR could have unacceptably high cardinality.
+
+There are three dimensions that determine the cardinality of HHR.
+1. The number of ranges we collect metrics for.
+2. The sample rate, i.e. how often we collect metrics.
+3. The information captured for each range during sampling.
+
+Clusters running production workloads could have 100K to 1M ranges. At any sort of reasonable sample rate, this would result in too much data captured. To reduce the number of ranges we collect metrics for, we can set a budget of 1000 max ranges. To populate these 1000 ranges we can use a mixed strategy of filtering out the coldest ranges and aggregating what remains. This is discussed in detail later on.
+
+To provide meaningful resolution, we can set a sampling rate of 4 times per hour, and store samples for a period of 14 days.
+
+For each range, here is the information captured:
+| Value      | Bytes |  description |
+| ---------- |------ | ---      |
+| node_id    | 2    |   | 
+| store_id   | 2    |  |
```

Comment:
I added a point in the `Limitations` section that discusses this.  Per-replica metrics would be ideal. It might be best to let the consumer decide how to come up with an aggregate for the range.